spec/graphql/schema/printer_spec.rb

```diff
@@ -691,9 +691,9 @@ def self.visible?(member, ctx)
             when "SCALAR"
               true
             when "OBJECT", "UNION", "INTERFACE"
-              ctx[:names].include?(member.graphql_name)
+              ctx[:names].include?(member.graphql_name) || member.introspection?
             else
-              false
+              member.introspection?
```

Comment:
TODO: why this change? Same question about rake_test_spec.rb

Reply:
`Warden` didn't send root introspection types through `.visible?` checks when using the `#reachable_types` method, hence this difference. See also https://github.com/rmosolgo/graphql-ruby/pull/5018#discussion_r1686746510